router/pods_test.go

```diff
@@ -269,6 +269,54 @@ func TestGetRoutesInvalidPublicPathsPath(t *testing.T) {
 			PodIP: "1.2.3.4",
 		},
 	}))
+
+	// "%ZZ" is not a valid path for targetPath segment
+	validateRoutes(t, "pod has an invalid routingPaths path", []*Route{}, GetRoutes(config, &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Annotations: map[string]string{
+				config.PodsPathsAnnotation: genRoutes(path("/", "3000", "test")),
+			},
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				api.Container{
+					Ports: []api.ContainerPort{
+						api.ContainerPort{
+							ContainerPort: int32(3000),
+						},
+					},
+				},
+			},
+		},
+		Status: api.PodStatus{
+			Phase: api.PodRunning,
+			PodIP: "1.2.3.4",
+		},
+	}))
+
+	// "%ZZ" is not a valid path for targetPath segment
+	validateRoutes(t, "pod has an invalid routingPaths path", []*Route{}, GetRoutes(config, &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Annotations: map[string]string{
+				config.PodsPathsAnnotation: genRoutes(path("/test/%2a/%", "3000", "")),
```

Comment:
Do you mind adding an example path with bad and good encoding in the middle of the string?  The issues being fixed by the new regex are:

1. Previous regex only ensured one match was found _(not that the full string matched)_
2. Previous regex only matched either a valid character _(singluar)_ or a valid URI encoded character _(singular)_

This test would not show that the regex was wrong.